src/stores/my-account.ts

```diff
@@ -82,6 +151,57 @@ export const useMyAccount = create<State & Actions>()((set, get) => ({
     }
     return address
   },
+  login: async () => {
+    let {
+      web3Auth,
+      initializeWeb3Auth,
+      getPrivateKey,
+      _syncSessionKey,
+      _subscribeEnergy,
+    } = get()
+    if (!web3Auth) {
+      web3Auth = await initializeWeb3Auth()
+    }
+    const web3authProvider = await web3Auth.connect()
+
+    set(() => ({
+      provider: web3authProvider,
+    }))
+
+    const privateKey = await getPrivateKey()
+    if (privateKey) {
+      const signer = await loginWithSecretKey(decodeSecretKey(privateKey))
+
+      /// We could ans should rely on the user profile as displayed by Web3Auth as it gives us a better idea of who logged in
+      /// Possible ways would be to use IPFS to update the logged in user on first login and using the web3Auth generated private key,
+      /// we could ensure only the private has access to update the IPFS profile.
+      /// For now, we use anonymouse profile for every logged in use.
+      // const user = await web3Auth.getUserInfo()
+
+      const randomPicture = createAvatar(bottts, {
+        size: 128,
+        seed: signer.address,
+      }).toDataUriSync()
+      const name = generateRandomName(signer.address)
+      const authenticatedUser = {
+        name: name,
+        profilePic: randomPicture,
+        address: signer.address,
+        encodedSecretKey: privateKey,
+      } as AuthenticatedUser
```

Comment:
if we currently still use the anonymous for user info, then I think these lines are not needed, just make the authUser as null for now.

Reply:
Removed the authUser object already. we can think of user info later.